lib/constructs/apigateway.ts

```diff
@@ -7,7 +7,7 @@ import * as ec2 from "aws-cdk-lib/aws-ec2";
 import * as statement from "cdk-iam-floyd";
 
 import { Construct } from "constructs";
-import { HttpMethod } from "aws-cdk-lib/aws-events";
+type HttpMethod = "GET" | "HEAD" | "POST" | "PUT" | "DELETE" | "OPTIONS";
```

Comment:
There is an `HttpMethod` `enum` defined in `lib/http.ts`  that might be able to be used here if an `enum` is okay to use here.